File: clients/windows/Mailcal.Tests/AvatarFixture.cs
// An avatar for a fixture that does not draw one.
//
// The letters and the colour are the core's to derive, and every suite that needs one here is about
// something else, the auto-advance, a merged row, which row the list highlights. One stub keeps
// that decision out of them. The Android twin is AvatarFixture.kt.

using Allodia.Mailcal.ViewModels;
using uniffi.mailcal_bindings;

namespace Allodia.Mailcal.Tests;

internal static class AvatarFixture
{
    /// <summary>The core's avatar record, in the shape the FFI hands one over.</summary>
    internal static Avatar Core(string initials = "SR") =>
        new(Initials: initials,
            Light: new(Background: "#4C6EF5", Text: "#FFFFFF", Border: "#3B5BDB"),
            Dark: new(Background: "#4C6EF5", Text: "#FFFFFF", Border: "#3B5BDB"),
            ImagePath: null);

    /// <summary>The same avatar, projected into what the views bind.</summary>
    internal static AvatarItem Item(string initials = "SR") => AvatarItem.From(Core(initials));
}

File: clients/android/app/src/test/java/eu/allodia/mailcal/AvatarFixture.kt
package eu.allodia.mailcal

import uniffi.mailcal_bindings.Avatar
import uniffi.mailcal_bindings.Swatch

/**
 * A fixed avatar for fixtures whose tests do not care which person is drawn.
 *
 * The letters and the colour are the core's to derive, and every suite here is about something
 * else, a swipe, a quote, a merged row. One stub keeps that decision out of them.
 */
fun stubAvatar(initials: String = "SR"): Avatar = Avatar(
    initials = initials,
    light = Swatch(background = "#4C6EF5", text = "#FFFFFF", border = "#3B5BDB"),
    dark = Swatch(background = "#4C6EF5", text = "#FFFFFF", border = "#3B5BDB"),
    imagePath = null,
)
